app/controllers/api/v2/auth_source_ldaps_controller.rb

```diff
@@ -54,12 +54,31 @@ def update
         process_response @auth_source_ldap.update_attributes(params[:auth_source_ldap])
       end
 
+      api :PUT, "/auth_source_ldaps/:id/test/", N_("Test LDAP connection")
+      param :id, String, :required => true
+
+      def test
+        test = @auth_source_ldap.test_connection
```

Comment:
As per the comment in the test, you may actually have to consider returning a successful HTTP response, but with varying content depending on the success/failure of the test, rather than any 4xx or 5xx error code.  Otherwise REST clients, such as those in Hammer might simply treat it as an error either way.